### ISSUE TITLE
How to deploy to Vercel?

### ISSUE THREAD

**Issue by lionelrudaz** [ember-fastboot/ember-cli-fastboot#782]
Hi everybody, 

Thanks a lot for your wonderful work with Fastboot. 

I've been uploading a couple of apps on Heroku in the past, and now I have to do the same with Vercel.

I checked whether there was some blog posts or howto's and I could only find this: https://gist.github.com/oskarrough/42cef880cbfa874637e90c08102f18d0

The thing is that it's really outdated, some links are broken and it requires plenty of manual work.

I was wondering if any of you knows how to deploy on app on Vercel with Fastboot enabled. I've tried this afternoon, no luck yet. I don't know where to start TBH. I asked on Twitter, and got the answer that it's technically possible to do so.

Now, where should I start? 

- I've installed ember-cli-fastboot
- I've installed Fastboot-app-server
- I've created a file fastboot-server.js in the root or my project

```
const FastBootAppServer = require('fastboot-app-server');

const MY_GLOBAL = 'MY GLOBAL';

let server = new FastBootAppServer({
  distPath: 'dist',
  gzip: true, // Optional - Enables gzip compression.
  host: '0.0.0.0', // Optional - Sets the host the server listens on.
  port: 3000, // Optional - Sets the port the server listens on (defaults to the PORT env var or 3000).
  sandboxGlobals: { GLOBAL_VALUE: MY_GLOBAL }, // Optional - Make values available to the Ember app running in the FastBoot server, e.g. "MY_GLOBAL" will be available as "GLOBAL_VALUE"
  chunkedResponse: true // Optional - Opt-in to chunked transfer encoding, transferring the head, body and potential shoeboxes in separate chunks. Chunked transfer encoding should have a positive effect in particular when the app transfers a lot of data in the shoebox.
});

server.start();
```

Locally, it's working both with the general ember serve command and by launching fastboot-server.

Can you help me find my way? I might miss some basic knowledge. How do I deploy my server file on its own with Vercel? Is it a build task?

I'm happy if I can contribute to the community by writing a few lines on the subject in a readme here or elsewhere.

Let me know if you need more information.

Cheers,

Lionel

**Comment by perottilds** [comment]
Hey @lionelrudaz 


Sorry you had to wait for so long for a response. I've been fiddling with both vercel and ember lately. My hard guess would be to follow fastboot's "with node" option, for Vercel does not support "continuous runtime" (think EC2), only lambda functions

You could use only [one catch all function](https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes) as per NextJS: [...path].js or [...path].ts

So the project roughly look like:
```shell
something
├── api
│   └── [...fastboot].js
└── package.json
```

You would need to [rewrite requests](https://vercel.com/docs/configuration#project/rewrites) in vercel.json which has to be in the output folder as well I think as lambdas default to /api/*
```json
{
  "rewrites": [
    { "from": "/api/:paths*", "to": "/:paths*" }
  ]
}
```
Just make sure the output folder from the Vercel project configuration has this `api` folder.
For development, `vercel dev`, it runs the lambdas locally using the same configuration. You can change the port with the flag `--listen`

**Comment by lionelrudaz** [comment]
Thanks @perottilds for your answer. I've tried other attempts to upload the fastboot-server file to Vercel, but none worked. It turns out that Vercel is just not made for this architecture and doesn't allow the necessary customization to make it work.

 So at the end of the day, I switched to Heroku and moved my app over there. I keep Vercel for deploying some other apps running NextJS, it's a delight. I would consider it as well for a regular Ember app, but if you truly need server side rendering, it's not an option.
